### PR TITLE
Tensor LineSet

### PR DESCRIPTION
**Pull request by ssheorey**
- Basic functionality. Follows new naming scheme from PR #3855 .
- Python bindings.
- C++ tests.
- `tTriangleMesh::Clone()` bugfix: Clone() did not deep copy data. (missing unit tests in `tTriangleMesh`).
- `ToString()` C++ test for `tPointCloud` .
- Enable MATHJAX in Doxygen to display LaTeX Maths properly (e.g.: matrices).

---
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d/3867)

### CODE REVIEW

**Review by yxlao — CHANGES_REQUESTED** [review]
Looks good to me in general. Mostly cosmetic changes are needed.

> Reviewed 14 of 16 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d/3867#-:-MgOOkkL9qRXrD9v3wvU:b-c810sl)* status: all files reviewed, 5 unresolved discussions (waiting on @errissa and @ssheorey)

___
*[-- commits, line 1 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgO4rd66jrSeFXVvK6V:-MgO4rd66jrSeFXVvK6W:b-6nzv6e) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/-- commits#L1)):*
Remove this?
___
*[cpp/open3d/t/geometry/LineSet.h, line 58 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgO4u1v4lzLzPeQEFsF:-MgO4u1w70c1l-1iXmTq:b-kt4nwa) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L58)):*
> ```objc
> /// attributes have different levels:
> ///
> /// - Level 0: Default attribute {"positions", "indices"}.
> ```

I found Level 0, Level 1, ... less intuitive, so in #3855 they are just called "Default attribute", "Common attribute" and "Custom attribute". Also, some of the wordings have been updated. Could you update this to be consistent? 
___
*[cpp/open3d/t/geometry/LineSet.h, line 64 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgO5EYO9SKtVPYxyqyg:-MgO5EYO9SKtVPYxyqyh:b4f8m2l) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L64)):*
> ```objc
> ///     - Convenience functions:
> ///         - LineSet::GetPointPositions()
> ///         - LineSet::SetPointPositions(points_tensor)
> ```

remove `_tesnsor`, same for the rest, this was updated in #3855 
___
*[cpp/open3d/t/geometry/LineSet.h, line 73 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgO5IQHCN8kUljP0dJX:-MgO5IQHCN8kUljP0dJY:bastg5q) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L73)):*
> ```objc
> /// - Level 1: Commonly used attributes: line colors.
> ///     - Not created by default.
> ///     - The tensor must be of shape N x 3.
> ```

use `{}` for shape, e.g. `{N, 3}`, this was updated in #3855 
___
*[cpp/open3d/t/geometry/LineSet.h, line 118 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgOHXsg3gzJk8Z1qiUH:-MgOHXsg3gzJk8Z1qiUI:bd0zd20) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L118)):*
> ```objc
>     /// \param points A tensor with element shape (3,).
>     /// \param lines A tensor with element shape (2,) and Int Dtype.
>     LineSet(const core::Tensor &points, const core::Tensor &lines);
> ```

`point_positions`, `line_indices`

**Review by errissa — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d/3867)* status: all files reviewed, 6 unresolved discussions (waiting on @ssheorey)

___
*[cpp/open3d/t/geometry/LineSet.h, line 76 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgQh82R7sMPB1u47QT6:-MgQh82R7sMPB1u47QT7:bu4tfd9) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L76)):*
> ```objc
> ///     - Convenience functions:
> ///         - Line colors (stored at line_attr_["colors"])
> ///             - LineSet::GetLineColors()
> ```

The current shaders for visualization use per-vertex (or in the terminology here, per-point) colors. In other words, color would be an attribute of the points.

**Review by ssheorey — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d/3867)* status: 10 of 17 files reviewed, 6 unresolved discussions (waiting on @errissa and @yxlao)

___
*[-- commits, line 1 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgO4rd66jrSeFXVvK6V:-MgRPXxw6Hs-_2_2Stlt:bh0updz) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/-- commits#L1)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

Remove this?
</blockquote></details>

This looks like a new feature of reviewable. There isn't a file named `Commits` in the PR. I made the same comment for @errissa 's  Materials PR :-)
___
*[cpp/open3d/t/geometry/LineSet.h, line 58 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgO4u1v4lzLzPeQEFsF:-MgRZtE26QIOq4dStCBd:b-896fix) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L58)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

I found Level 0, Level 1, ... less intuitive, so in #3855 they are just called "Default attribute", "Common attribute" and "Custom attribute". Also, some of the wordings have been updated. Could you update this to be consistent? 
</blockquote></details>

Done.
___
*[cpp/open3d/t/geometry/LineSet.h, line 64 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgO5EYO9SKtVPYxyqyg:-MgRdhNd3u_thY6wFuwg:b-896fix) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L64)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

remove `_tesnsor`, same for the rest, this was updated in #3855 
</blockquote></details>

Done.
___
*[cpp/open3d/t/geometry/LineSet.h, line 73 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgO5IQHCN8kUljP0dJX:-MgRdi8xBLWPZZ9pF0HO:b-896fix) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L73)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

use `{}` for shape, e.g. `{N, 3}`, this was updated in #3855 
</blockquote></details>

Done.
___
*[cpp/open3d/t/geometry/LineSet.h, line 76 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgQh82R7sMPB1u47QT6:-MgRdvtL6pqPPtIr6zpG:bbd8vca) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L76)):*
<details><summary><i>Previously, errissa (Rene) wrote…</i></summary><blockquote>

The current shaders for visualization use per-vertex (or in the terminology here, per-point) colors. In other words, color would be an attribute of the points.
</blockquote></details>

@errissa how do things work if two points joined by a line have different colors?
___
*[cpp/open3d/t/geometry/LineSet.h, line 118 at r2](https://reviewable.io/reviews/isl-org/open3d/3867#-MgOHXsg3gzJk8Z1qiUH:-MgRf8tqDXvR3cu7HJe7:b-896fix) ([raw file](https://github.com/isl-org/open3d/blob/c6b133416d083b5b4b7c5b44b4c87a2090619648/cpp/open3d/t/geometry/LineSet.h#L118)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

`point_positions`, `line_indices`
</blockquote></details>

Done.

**Review by yxlao — COMMENTED** [review]
> Reviewed 6 of 7 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d/3867)* status: all files reviewed, 2 unresolved discussions (waiting on @errissa and @ssheorey)

___
*[cpp/pybind/t/geometry/lineset.cpp, line 55 at r4](https://reviewable.io/reviews/isl-org/open3d/3867#-MgdxvmQAk_uWH152VJp:-MgdxvmQAk_uWH152VJq:b-3uikpk) ([raw file](https://github.com/isl-org/open3d/blob/9c794229d35dc26fa9d4f96fae95d2c3b6aead1d/cpp/pybind/t/geometry/lineset.cpp#L55)):*
> ```cpp
> ``points["positions"]``.
> 
> Although the attributes are all stored in a key-value pair dictionary, the
> ```

Giving code examples might be more intuitive here. 

I've updated the point cloud/triangle mesh counterparts as well:

https://github.com/isl-org/Open3D/blob/361307af1163917fe3ef48df738d6af14c55cb73/cpp/pybind/t/geometry/trianglemesh.cpp#L43-L95

**Review by ssheorey — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d/3867)* status: 11 of 17 files reviewed, 2 unresolved discussions (waiting on @errissa and @yxlao)

___
*[cpp/pybind/t/geometry/lineset.cpp, line 55 at r4](https://reviewable.io/reviews/isl-org/open3d/3867#-MgdxvmQAk_uWH152VJp:-Mgh0P7c85R-7GC17KZl:b-896fix) ([raw file](https://github.com/isl-org/open3d/blob/9c794229d35dc26fa9d4f96fae95d2c3b6aead1d/cpp/pybind/t/geometry/lineset.cpp#L55)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

Giving code examples might be more intuitive here. 

I've updated the point cloud/triangle mesh counterparts as well:

https://github.com/isl-org/Open3D/blob/361307af1163917fe3ef48df738d6af14c55cb73/cpp/pybind/t/geometry/trianglemesh.cpp#L43-L95

</blockquote></details>

Done.

**Review by yxlao — APPROVED** [review]

File: cpp/tests/t/geometry/LineSet.cpp [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    EXPECT_TRUE(lineset.IsEmpty());
+    EXPECT_FALSE(lineset.HasPointPositions());
+    EXPECT_FALSE(lineset.HasLineIndices());
+    EXPECT_FALSE(lineset.HasLineColors());
+
+    // Default device.
+    EXPECT_EQ(lineset.GetDevice(), core::Device("CPU:0"));
+
+    // ToString
+    EXPECT_EQ(lineset.ToString(), R"(LineSet on CPU:0
```
Comment (reyanshsolis) [inline cpp/tests/t/geometry/LineSet.cpp:67]: We may also use `EXPECT_STREQ(lineset.ToString().c_str(), text.c_str());`

**Pull request merged**